src/Generator/ClassGenerator.php

```diff
@@ -87,6 +88,11 @@ public static function fromReflection(ClassReflection $classReflection)
         }
 
         $cg->setAbstract($classReflection->isAbstract());
+        $cg->setFinal($classReflection->isFinal());
```

Comment:
@Ocramius this call was missing before - not sure if that was intentional. If not, I think we should backport it to 4.7.x too.

Reply:
Probably best to patch `4.7.x` with a dedicated test too

Reply:
Done in #156